src/html5sortable.ts

```diff
@@ -512,13 +511,12 @@ export default function sortable (sortableElements, options) {
           _before(element, placeholder)
         }
         // Intentionally violated chaining, it is more complex otherwise
-        placeholders
+        Array.from(placeholderMap.values())
           .filter(function (element) { return element !== placeholder })
           .forEach(_detach)
-      } else {
-        if (placeholders.indexOf(element) === -1 &&
+        if (Array.from(placeholderMap.values()).indexOf(element) === -1 &&
```

Comment:
And couldn't you use `placeholderMap.get(sortableElement) !== element`? Would that work, I did not really use Map yet.

Reply:
I assume you mean for
```
if (Array.from(placeholderMap.values()).indexOf(element) === -1 && !_filter(_getChildren(element), options.items).length)
```
yeah I'm not sure this line confuses me. I'm not sure what is happening there.

Reply:
So this part `Array.from(placeholderMap.values()).indexOf(element) === -1` check for element NOT existing in the array.

Reply:
correct. I have no idea what this does? Length of 0 == true? everything else is false?
```
!_filter(_getChildren(element), options.items).length
```

Reply:
Yes, this checks if the list (element) is empty  (= has no child of the type specified as option.items) in it. Because if the list is empty we need to `element.appendChild(placeholder);` Otherwise we insertAfter or Before the hovered item.

This code can probably written in a better and easier to read way, though.